stock_picking_sale_order_link/models/stock_picking.py

```diff
@@ -12,4 +12,7 @@ def action_view_sale_order(self):
         of given picking.
         """
         self.ensure_one()
-        return self.sale_id.get_formview_action()
+        # Remove default_picking_id to avoid defaults get
+        # https://github.com/odoo/odoo/blob/e4d22d390c8aa8edf757e36704a9e04b2b89f115/
+        # addons/stock/models/stock_move.py#L410
+        return self.with_context(default_picking_id=False).sale_id.get_formview_action()
```

Comment:
Better to simply remove the key as others do, as sometimes the inexistence of the key is not the same than having it with False:
```suggestion
        ctx = self.env.context.copy()
        ctx = ctx.pop(default_picking_id, False)
        return self.with_context(ctx).sale_id.get_formview_action()
```